### ISSUE TITLE
More precision on the classification side in object detection

### ISSUE THREAD

**Issue by zpmmehrdad** [AlexeyAB/darknet#5086]
Hi @AlexeyAB ,

I have ~320 classes and my classes are traffic signs. I'm going to use a model in real-time. Sometimes it detected wrong especially on speed limit signs and some signs that are similar in shape. I tested csresnext50-panet-spp-original-optimal. The result was good but when I test on a new videos that some signs are not in my classes the model is detecting the sign 100% if it is wrong. I want to reduce the false.
Could you please recommend me for this case?

For example, these are some of the signs that are slightly different. I detected all of them but it's not classified correctly. I think the problem is in classification side.

![120px_W4-1 svg](https://user-images.githubusercontent.com/5433346/77250492-f4a25f00-6c65-11ea-9b3a-ff5f4f180ce9.png)               ![120px_W4-3 svg](https://user-images.githubusercontent.com/5433346/77250510-0e43a680-6c66-11ea-8d03-4cab5d4934f2.png)

![Australia_R2-9_(R) svg](https://user-images.githubusercontent.com/5433346/77250528-3af7be00-6c66-11ea-9e5f-6837c34e9440.png)               ![Australia_R2-9_(L) svg](https://user-images.githubusercontent.com/5433346/77250531-40550880-6c66-11ea-8c78-8330092e456b.png)

Thanks

**Comment by AlexeyAB** [comment]
Show screenshots of results with wrong detections

**Comment by zpmmehrdad** [comment]
@AlexeyAB Hi,

For example in this picture, I have another sign that is not in my dataset but it detected "School" with 98% accuracy that it's not correct. I am sure about the dataset because I checked it several times.

![111](https://user-images.githubusercontent.com/5433346/77322563-8dea7780-6d31-11ea-8fc1-65c037aae30c.jpg)


Thanks

**Comment by zpmmehrdad** [comment]
@AlexeyAB Hi,

In this [repo](https://github.com/mohamedameen93/German-Traffic-Sign-Classification-Using-TensorFlow), he could get 99% accuracy on validation set but use the VGG model for classification, not detection. How can I achieve that?

**Comment by AlexeyAB** [comment]
* What mAP do you get on Validation dataset?

> I have another sign that is not in my dataset but it detected "Children Sign" with 98% accuracy that it's not correct. 

* Read: https://github.com/AlexeyAB/darknet#how-to-improve-object-detection

> desirable that your training dataset include images with non-labeled objects that you do not want to detect - negative samples without bounded box (empty .txt files) - use as many images of negative samples as there are images with objects

**Comment by zpmmehrdad** [comment]
@AlexeyAB Hi,

I got 90% on validation dataset. Unfortunately when I test on new videos I get wrong result and I don't have them all, I have to test new vidoes first to find the wrong detection. I want to decrease the accuracy on wrong detection at least. In those signs that I show you can I get good result about the classification?

**Comment by AlexeyAB** [comment]
> but it detected "Children Sign" with 98% **accuracy**

This is not accuracy, this is confidence.

>  In those signs that I show you can I get good result about the classification?

Use this approach: https://github.com/AlexeyAB/darknet/issues/5086#issuecomment-602732744

**Comment by zpmmehrdad** [comment]
@AlexeyAB Hi,
I saw it before and I used that. I wanted to know, is there any way? Example, change the loss function. I asked @WongKinYiu and he said that you should use triplet loss.

**Comment by AlexeyAB** [comment]
Triplet loss is in the roadmap: https://github.com/AlexeyAB/darknet/issues/4972
